core/src/main/codegen/templates/Parser.jj

```diff
@@ -2173,10 +2176,23 @@ SqlNode TableRef2(boolean lateral) :
         <TABLE> { s = span(); } <LPAREN>
         tableRef = TableFunctionCall(s.pos())
         <RPAREN>
+        [
+            <HINT_BEG>
+            CommaSeparatedSqlHints(hints)
+            <COMMENT_END>
+        ]
         {
             if (lateral) {
-                tableRef = SqlStdOperatorTable.LATERAL.createCall(
-                    s.end(this), tableRef);
+                if (hints.size() > 0) {
+                    hintList = new SqlNodeList(hints, s.addAll(hints).end(this));
+                    SqlNode tmpTableRef = SqlStdOperatorTable.LATERAL.createCall(
+                                        s.end(this), tableRef);
```

Comment:
`tmpTableRef ` => `lateralCall`

Reply:
I feel it is more reasonable to include the `tableRef` into the `LATERAL` call not the other way around, say we should change the lateral call to support `tableRef` as the table reference, WDYT ?

Reply:
I've already changed it